core/src/banking_stage.rs

```diff
@@ -480,15 +481,43 @@ impl BankingStage {
         Ok((chunk_start, unprocessed_txs))
     }
 
-    fn process_received_packets_using_closure<'a, F>(
+    fn filter_out_invalid_transactions(
```

Comment:
This function can use a couple tests and a code comment.  Probably a better name too (`filter_valid_transaction_indexes`?)

Reply:
Reworked the function and added some tests.